laws/src/main/scala/algebra/laws/RingLaws.scala

```diff
@@ -107,7 +107,15 @@ trait RingLaws[A] extends GroupLaws[A] {
     name = "ring",
     al = additiveCommutativeGroup,
     ml = multiplicativeMonoid,
-    parents = Seq(rig, rng)
+    parents = Seq(rig, rng),
+    "fromInt" -> forAll { (n: Int) =>
+      Ring.fromInt[A](n) ?== A.sumN(A.one, n)
+    },
+    "fromBigInt" -> forAll { (ns: List[Int]) =>
```

Comment:
Exactly - we do have a homomorphism here, but I don't think the law is too strong. Any ring has a homomorphism from Z -> R (BigInt -> A) and this is really just codifying this.